applications/CompressiblePotentialFlowApplication/custom_processes/replace_elements_and_conditions_for_adjoint_problem_process.h

```diff
@@ -0,0 +1,163 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder
+//                   Armin Geiser
+//
+
+
+#if !defined(KRATOS_REPLACE_ELEMENTS_AND_CONDITIONS_ADJOINT_PROCESS_H_INCLUDED)
+#define  KRATOS_REPLACE_ELEMENTS_AND_CONDITIONS_ADJOINT_PROCESS_H_INCLUDED
+
+// System includes
+#include <string>
+
+// External includes
+
+// Project includes
+#include "includes/kratos_parameters.h"
+#include "processes/process.h"
+#include "includes/model_part.h"
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @brief This class replaces all elements and conditions of a structural
+ * model part with their adjoint counterparts. If this process is called on a
+ * primal model part, it is transformed to an adjoint model part and vice versa.
+ * The variable "IS_ADJOINT" at the model parts process info is set accordingly.
+ */
+class ReplaceElementsAndConditionsAdjointProcess : public Process
```

Comment:
this is copy-pased from Structuralmechanics, which is a bad idea

=> To me it seems like this class could be refactored to accept a parameters-object to specify the Primal-Adjoint names
Then it could be used for both cases

@armingeiser @MFusseder what do you think?


or even a different question, @marcnunezc do you have different elements in one modelpart? if not then you couldd use the process from the core directly

Reply:
With respect

> this is copy-pased from Structuralmechanics, which is a bad idea
> 
> => To me it seems like this class could be refactored to accept a parameters-object to specify the Primal-Adjoint names
> Then it could be used for both cases
> 
> @armingeiser @MFusseder what do you think?

If I'm not mistaken, they are planning to move this kind of utilities to the core in the following weeks!

>
> or even a different question, @marcnunezc do you have different elements in one modelpart? if not then you couldd use the process from the core directly

I do not have more than one now, but I think you and @inigolcanalejo are working on splitting the "incompressible_potential_flow_element" into several elements instead of using many "WAKE"/"KUTTA".. flags as the current version. Also, eventually this will be adapted to embedded elements which will also require more than one element per model part.

Reply:
ok then I see why you need it

@MFusseder @armingeiser we should speak abt how to generalize this Process such that it can be moved to the core

Reply:
i am working on this at the moment, I will let you know once it is done.